work11/index.js

```diff
@@ -0,0 +1,71 @@
+"use strict";
+const div = document.querySelector(".wrap")
+const ul = document.getElementById("js-list");
+const url = "https://jsondata.okiba.me/v1/json/d4ex5210822124352"
+
+function getData() {
+  const result = new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve(
+        async function () {
+          const response = (await fetch(url)).json();
+          return response
+        }()
+      )
+
+    }, 3000);
+  })
+  return result;
+}
+
+async function displayView() {
+  loading();
+
+  try {
+    const arrayData = await getData();
+    createElement(arrayData);
+  } catch (e) {
+    div.textContent = e.message;
+  } finally {
+    hideLoading();
+  }
+
+}
+
+function hideLoading() {
+  ul.style.backgroundImage = "none";
+}
+
+function createElement(imgArray) {
+  const fragment = document.createDocumentFragment();
+
+  Object.keys(imgArray).forEach(key => {
+
+    Object.keys(imgArray[key]).forEach(keyIndex => {
+
+      const li = document.createElement("li");
+      const a = document.createElement("a");
+      const img = document.createElement("img");
+
+      a.href = imgArray[key][keyIndex].a;
+      a.textContent = imgArray[key][keyIndex].text;
+      img.src = imgArray[key][keyIndex].img;
+      img.alt = imgArray[key][keyIndex].alt;
+
```

Comment:
IMO
It seems a little complicated...

I would like to know if there is a reason why you used Object.keys.
Also, if it were me, I'd write it like this
```suggestion
  Object.keys(imgArray).forEach(key => {
  
    imgArray[key].forEach(keyIndex => {
    
      const li = document.createElement("li");
      const a = document.createElement("a");
      const img = document.createElement("img");

      a.href = keyIndex.a;
      a.textContent = keyIndex.text;
      img.src = keyIndex.img;
      img.alt = keyIndex.alt;
```

Reply:
IMO
I learned division and assignment from Mr. Morita in this part.（分割代入）
```suggestion
 function createElement({ data }) {
  const fragment = document.createDocumentFragment();
  data.forEach((value) => {
    const li = document.createElement("li");
    const a = document.createElement("a");
    const img = document.createElement("img");
    a.href = value.a;
    a.textContent = value.text;
    img.src = value.img;
    img.alt = value.alt;
    fragment.appendChild(li).appendChild(a).prepend(img);
  });
  ul.appendChild(fragment);
}
```

Reply:
@443n0511 
Amazing...!!!!

I regret that I did not know about division and assignment until today.
I learned a lot.
Thank you for everything.

Reply:
The reason for using Object.key in the second line is "imgArray [key] .forEach (keyIndex => {...}".
I couldn't think of an explanation.

The reason is that I didn't understand the meaning of getting the key value of the object in Object.key.

I understand now.

commit number
822878e

Reply:
Thanks for the fix.

I had not learned enough about Object[key].
I will take this opportunity to study it. 

Also, I hope you will update codesandbox with every revision.
Thank you very much.

Reply:
@443n0511 

I'm sorry to be late.
Fixed "partitioned assignment(分割代入)".